### DISCUSSION TITLE
Option to mirror providers and configurations transitively

### DISCUSSION THREAD

**Discussion by reegnz** [crossplane/crossplane#4005] [Q&A]
I'd like to use crossplane in an AWS VPC with heavily restricted internet-access. That means only allowing internet egress to a few AWS services and that's it (eg. AWS-IAM or any other AWS service that does not support AWS PrivateLink VPC endpoints).

I imagine I can push them as they're OCI images, but is there a way to easily mirror crossplane configs with transitive provider dependencies?

**Comment by bobh66** [comment]
You can put them in ECR or deploy a local Harbor and store them there. 
Then deploy crossplane with the "-r repository" argument to specify the 
default repo where Crossplane should look for artifacts.

Bob

On 4/20/2023 7:54 AM, Zoltán Reegn wrote:
>
> I'd like to use crossplane in an AWS VPC with heavily restricted 
> internet-access. That means only restricting internet egress to a few 
> AWS services and thats it (eg. AWS-IAM or any other AWS service that 
> does not support AWS PrivateLink VPC endpoints).
>
> Is there a way to mirror crossplane configurations and providers into 
> something like an S3 bucket so they can be downloaded from there 
> instead of upbound's servers?
>
> —
> Reply to this email directly, view it on GitHub 
> <https://github.com/crossplane/crossplane/discussions/4005>, or 
> unsubscribe 
> <https://github.com/notifications/unsubscribe-auth/ABFU5CHXE6VPLXQLO7C5L7DXCEWZFANCNFSM6AAAAAAXFOYB2I>.
> You are receiving this because you are subscribed to this 
> thread.Message ID: 
> ***@***.***>
>


-- 
Bob Haddleton
Director of R&D Innovation, Advanced Technology Design Studio
Cloud and Network Services
Nokia
Contact number: +1 630 805 2990
